planner/core/exhaust_physical_plans.go

```diff
@@ -1612,6 +1647,74 @@ func (p *LogicalJoin) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]P
 	return joins, true
 }
 
+func (p *LogicalJoin) tryToGetMppHashJoin(prop *property.PhysicalProperty, useBCJ bool) []PhysicalPlan {
+	if !prop.IsEmpty() {
+		return nil
+	}
+	if prop.TaskTp != property.RootTaskType && prop.TaskTp != property.MppTaskType {
+		return nil
+	}
+
+	lkeys, rkeys, _, _ := p.GetJoinKeys()
```

Comment:
_, _, _, hasNullEQ := p.GetJoinKeys()
	if hasNullEQ {
		return nil
	}

also need?

Reply:
yeah my bad, still needs it

Reply:
where updated?